rules/cloud/azure/identity_protection/azure_identity_protectection_anomalous_token.yml

```diff
@@ -0,0 +1,22 @@
+title: Anomalous Token
+id: 6555754e-5e7f-4a67-ad1c-4041c413a007
+status: experimental
+description: Indicates that there are abnormal characteristics in the token such as an unusual token lifetime or a token that is played from an unfamiliar location.
+references:
+    - https://learn.microsoft.com/en-us/azure/active-directory/identity-protection/concept-identity-protection-risks#anomalous-token
+    - https://learn.microsoft.com/en-us/azure/active-directory/architecture/security-operations-user-accounts#unusual-sign-ins
+author: Mark Morowczynski '@markmorow'
+date: 2023/08/07
+tags:
+    - attack.t1528
+    - attack.credential_access
+logsource:
+    product: azure
+    service: riskDetection
+detection:
+    selection:
+        properties.message: anomalousToken
```

Comment:
From the links you posted, we assume the response will be parsed and the field will be `riskEventType` so it's best to use that imo instead of the full message

Reply:
I will defer to you and @frack113  on what you want. There is going to be about 20 of these so I'll make sure they are consistent. The alert type will change.

Reply:
The general logic of SIGMA is to use specific field that instead of full log messages. As generally the message is parsed and fields are searched directly. Since the fields are clear by the docs, I think its best to keep the convention.

cc @frack113

Reply:
From a resource point of view, general searches are slow and greedy.
I'd rather deactivate a rule than have to increase processing capacity for it.

You can have a look at powershell classic rules.
Even if Microsoft only provides a raw Key-Value string, the rules use the Key as the field name.
In logsource : `definition: fields have to be extract from event`.

With a field name that makes it easier for rookied SOC operators to read the rule.
It's easier to understand `door_state: open` than just `open` as a search.

Reply:
From https://learn.microsoft.com/en-us/graph/api/resources/riskdetection?view=graph-rest-1.0#riskeventtype-values
It is only `riskEventType` in the selection.

`riskDetail` seen to have a strange  behaving , and can be "none".

So only
```
selection:
    riskEventType: 'anomalousToken'
condition: selection
```

Reply:
I'm good with that. There will be many other types our team will make here and submit for you all. We'll follow this pattern.